hera_pspec/plot.py

```diff
@@ -0,0 +1,171 @@
+                   average_times=False, fold=False, plot_noise=False, 
+                   delay=True, deltasq=False, little_h=True, 
+                   legend=False, ax=None):
+    """
+    Plot a 1D delay spectrum for a group of baselines.
+    
+    Parameters
+    ----------
+    uvp : UVPspec
+        UVPSpec object, containing delay spectra for a set of baseline-pairs, 
+        times, polarizations, and spectral windows.
+    
+    blpairs : list of tuples or lists of tuples
+        List of baseline-pair tuples, or groups of baseline-pair tuples.
+    
+    spw, pol : int or str
+        Which spectral window and polarization to plot.
+    
+    average_blpairs : bool, optional
+        If True, average over the baseline pairs within each group.
+        
+    average_times : bool, optional
+        If True, average spectra over the time axis. Default: False.
+    
+    fold : bool, optional
+        Whether to fold the power spectrum in |k_parallel|. Default: False.
+    
+    plot_noise : bool, optional
+        Whether to plot noise power spectrum curves or not. Default: False.
+    
+    delay : bool, optional
+        Whether to plot the power spectrum in delay units (ns) or cosmological 
+        units (h/Mpc). Default: True.
+    
+    deltasq : bool, optional
+        If True, plot dimensionless power spectra, Delta^2. This is ignored if 
+        delay=True. Default: False.
+    
+    little_h : bool, optional
+        If using cosmological units (i.e. delay=False), whether to use h^-1 
+        units or not. Default: True.
+    
+    legend : bool, optional
+        Whether to switch on the plot legend. Default: False.
+    
+    ax : matplotlib.axes, optional
+        Use this to pass in an existing Axes object, which the power spectra 
+        will be added to. (Warning: Labels and legends will not be altered in 
+        this case, even if the existing plot has completely different axis 
+        labels etc.) If None, a new Axes object will be created. Default: None.
+    
+    Returns
+    -------
+    ax : matplotlib.axes
+        Matplotlib Axes instance.
+    """
+    # Create new Axes if none specified
+    new_plot = False
+    if ax is None:
+        new_plot = True
+        #ax = plt.subplot()
+        fig, ax = plt.subplots(1, 1)
+    
+    # Add ungrouped baseline-pairs into a group of their own (expected by the
+    # averaging routines) # FIXME: Check if this is an in-place operation
+    blpairs_in = blpairs
+    blpairs = [] # Must be a list, not an array
+    for i, blpgrp in enumerate(blpairs_in):
+        if not isinstance(blpgrp, list):
+            blpairs.append([blpairs_in[i],])
+        else:
+            blpairs.append(blpairs_in[i])
+    
+    # Average over blpairs or times if requested
+    blpairs_in = copy.deepcopy(blpairs) # Save input blpair list
+    if average_blpairs:
+        uvp_plt = uvp.average_spectra(blpair_groups=blpairs, 
```

Comment:
I think this could be as simple as 
```
uvp = uvp.average_spectra(..., inplace=False)
```
and then further down you could additionally do a time average as
```
uvp = uvp.average_spectra(time_avg=True, inplace=False)
```
do you need to form `blpair_list`? I think it does that for you in the function if its empty...

Also it seems like in this structure one cannot *both* average over baselines and times? It would be nice to have
```
if average_times:
   uvp = uvp.average_spectra(..., inplace=False)
if average_blpairs:
    uvp = uvp.average_spectra(..., inplace=False)
```
so that we could do both. Also, its better to have time averaging precede blpair averaging.

Reply:
I simplified it a little bit, thanks. It can already do both types of averaging at the same time -- if `average_blpairs` is True, it passes `time_avg=average_times` to `average_spectra()`.